src/Core/Framework/DataAbstractionLayer/Dbal/Common/RepositoryIterator.php

```diff
@@ -56,6 +58,23 @@ public function __construct(
         $this->context = clone $context;
     }
 
+    public function reset(): void
```

Comment:
Maybe name it `rewind` ? https://www.php.net/manual/en/function.rewind.php That's what php use for such things.

A simple php doc would also be nice:

Rewinds the Iterator to the beginning

Reply:
If I start calling it rewind, others would expect other iterator methods, right? I mean I can rename it but I assume, that this is creating confusion